lib/models/user-identity.js

```diff
@@ -112,14 +114,23 @@
       var userModel = (userIdentityModel.relations.user &&
                        userIdentityModel.relations.user.modelTo) ||
                        loopback.getModelByType(loopback.User);
-      var userObj = (options.profileToUser || profileToUser)(provider, profile);
-      if (!userObj.email) {
-        return cb('email is missing from the user profile');
+      var userObj = (options.profileToUser || profileToUser)(provider, profile, options);
+      if (!userObj.email && !options.emailOptional) {
+        process.nextTick(function() {
```

Comment:
We're integrating loopback-component-passport into our Loopback app and are having some trouble understanding this code. Why is the `nextTick` required here - it seems like it allows the execution to fall through and to create the member even if the email is not available.